### DISCUSSION TITLE
[General]

### DISCUSSION THREAD

**Discussion by jpnr79** [glpi-project/glpi-agent#407] [General]
### Your question

SQL Instace brings the name of SQL version and not sql instance, on server that haave multiple instance that is bad, i offer help if you want to do it right

**Comment by g-bougard** [comment]
Hi @jpnr79 

of course, if you find a bug and have the skill to help us to understand, we will take it.

Can you share the output of the following command run from an administrative console and from the GLPI-Agent agent folder ? Only the `databases_services` node is relevant.
```
glpi-inventory --partial database
```
Then can you share what output you expected or point which values are wrong ? And finally can you give me a method to catch the right value ?

**Comment by jpnr79** [comment]
Hello Guillaume

I'm going to send you the information, and help to make this part great.

I have another question and problem in hands, that is netdiscovery dont
resolve name, there are any way to put nsloockup to give the name when
netdiscovery dont resolve name by netbios, by arp or snmp?

Guillaume Bougard ***@***.***> escreveu no dia segunda,
29/05/2023 à(s) 08:51:

> Hi @jpnr79 <https://github.com/jpnr79>
>
> of course, if you find a bug and have the skill to help us to understand,
> we will take it.
>
> Can you share the output of the following command run from an
> administrative console and from the GLPI-Agent agent folder ? Only the
> databases_services node is relevant.
>
> glpi-inventory --partial database
>
> Then can you share what output you expected or point which values are
> wrong ? And finally can you give me a method to catch the right value ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glpi-project/glpi-agent/discussions/407#discussioncomment-6027295>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGJFAZ5ZYEHOAXKQ4L5HLKDXIRIP5ANCNFSM6AAAAAAYRCLFXI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Pedro Rocha
Faro
email: ***@***.*** ***@***.***>
Telf.: +351 935 478 010 <+351%20935%20478%20010>
linkedin: Pedro Rocha <http://www.linkedin.com/in/pedro-rocha79>

**Reply by g-bougard** [reply]
Please open another discussion for that second question.

**Reply by g-bougard** [reply]
Okay, you still did in #406... missed it sorry

**Comment by jpnr79** [comment]
Hello Guillaume

This is the out put you ask, the problem is the "name":"SQL SERVER 2019"
the real name for the instance is MSSQLSERVER that is the default

C:\Program Files\GLPI-Agent>glpi-inventory.bat --partial database
[info] New partial inventory from
NSRBDF01.justica.local-2023-05-29-09-26-30 for local0
{
   "action": "inventory",
   "content": {
      "bios": {
         "assettag": "8557-7311-0707-8618-3869-1851-25",
         "bdate": "2019-12-17",
         "biosserial": "8557-7311-0707-8618-3869-1851-25",
         "bmanufacturer": "Microsoft Corporation",
         "bversion": "Hyper-V UEFI Release v4.0",
         "enclosureserial": "8557-7311-0707-8618-3869-1851-25",
         "mmodel": "Virtual Machine",
         "msn": "8557-7311-0707-8618-3869-1851-25",
         "smanufacturer": "Microsoft Corporation",
         "smodel": "Virtual Machine",
         "ssn": "8557-7311-0707-8618-3869-1851-25"
      },
      "databases_services": [
         {
            "databases": [
               {
                  "creation_date": "2003-04-08 09:13:36",
                  "is_active": true,
                  "name": "master",
                  "size": 519,
                  "update_date": "2023-05-11 15:42:15"
               },
               {
                  "creation_date": "2023-05-10 08:50:46",
                  "is_active": true,
                  "name": "tempdb",
                  "size": 5120,
                  "update_date": "2023-05-29 09:25:00"
               },
               {
                  "creation_date": "2003-04-08 09:13:36",
                  "is_active": true,
                  "name": "model",
                  "size": 768,
                  "update_date": "2020-11-12 10:43:19"
               },
               {
                  "creation_date": "2019-09-24 14:21:42",
                  "is_active": true,
                  "name": "msdb",
                  "size": 1061,
                  "update_date": "2023-03-31 14:07:12"
               },
               {
                  "creation_date": "2020-11-12 12:02:50",
                  "is_active": true,
                  "name": "M3Database",
                  "size": 3648,
                  "update_date": "2021-05-24 23:56:25"
               },
               {
                  "creation_date": "2022-12-20 09:37:27",
                  "is_active": true,
                  "name": "AdventureWorks2019",
                  "size": 592,
                  "update_date": "2023-05-11 15:22:07"
               },
               {
                  "creation_date": "2022-03-17 11:08:29",
                  "is_active": true,
                  "name": "DBA",
                  "size": 1024,
                  "update_date": "2023-02-03 11:23:39"
               },
               {
                  "creation_date": "2023-03-23 15:44:09",
                  "is_active": true,
                  "name": "teste_xpto_erro_viste",
                  "size": 1024,
                  "update_date": "2023-05-11 15:59:39"
               }
            ],
            "is_active": true,
            "last_boot_date": "2023-05-10 08:50:42",
            "manufacturer": "Microsoft",
            "name": "SQL Server 2019",
            "port": 1433,
            "size": 13757,
            "type": "mssql",
            "version": "15.0.4236.7"
         }
      ],
      "hardware": {
         "chassis_type": "Desktop",
         "memory": 8190,
         "name": "NSRBDF01",
         "uuid": "B3CA54D3-3D17-4697-B9CF-A96546846FDA",
         "vmsystem": "Hyper-V",
         "winlang": "1033",
         "winowner": "Windows User",
         "winprodid": "00429-70000-00000-AA596",
         "winprodkey": "XXXXX-XXXXX-XXXXX-XXXXX-J464C",
         "workgroup": "justica.local"
      },
      "versionclient": "GLPI-Inventory_v1.4"
   },
   "deviceid": "NSRBDF01.justica.local-2023-05-29-09-26-30",
   "itemtype": "Computer",
   "partial": true
}

**Reply by g-bougard** [reply]
The `name` is extracted from the output of `SELECT @@version` SQL request. So how I can find "MSSQLSERVER" from a SQL request ?

**Comment by jpnr79** [comment]
you can do a select @@servername

Show you example from a server
[image: image.png]
[image: image.png]


Guillaume Bougard ***@***.***> escreveu no dia segunda,
29/05/2023 à(s) 09:55:

> The name is extracted from the output of SELECT @@version SQL request. So
> how I can find "MSSQLSERVER" from a SQL request ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glpi-project/glpi-agent/discussions/407#discussioncomment-6027755>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGJFAZ3LUJGLHJBFHWT3FN3XIRQBTANCNFSM6AAAAAAYRCLFXI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Pedro Rocha
Faro
email: ***@***.*** ***@***.***>
Telf.: +351 935 478 010 <+351%20935%20478%20010>
linkedin: Pedro Rocha <http://www.linkedin.com/in/pedro-rocha79>

**Reply by g-bougard** [reply]
Hi @jpnr79 
please don't use email to answer github notification. There's a lot of pollution as you include email history and images are not displayed. So better answer from the site.

**Comment by jpnr79** [comment]
Now i send you a powershell that give you information based on regedit,
this give you all instances on a server and for each the information of the
port, patch version, edition, version etc..

$obj = [PSCustomObject] @{
    data = @((get-itemproperty 'HKLM:\SOFTWARE\Microsoft\Microsoft SQL
Server').InstalledInstances | % {
      [PSCustomObject] @{
'AgentDisplayName' = $_
            'AgentServiceName' = if($_ -eq 'MSSQLSERVER') {
"SQLSERVERAGENT" } else { "SQLAgent`$$_" }
            'SqlPerfmonN' = if($_ -eq 'MSSQLSERVER') { "SQLServer" } else {
"MSSQL`$$_" }
'AgentPerfmon' = if($_ -eq 'MSSQLSERVER') { "SQLAgent" } else {
"SQLAgent`$$_" }
            'InstanceDisplayName' = $_
'InstanceServiceName' = if($_ -eq 'MSSQLSERVER') { "MSSQLSERVER" } else {
"MSSQL`$$_" }
            'Path' = $path = (Get-ItemProperty
'HKLM:\SOFTWARE\Microsoft\Microsoft SQL Server\Instance Names\SQL').$_
            'InstancePort'= (Get-ItemProperty
"HKLM:\SOFTWARE\Microsoft\Microsoft SQL
Server\$path\MSSQLServer\SuperSocketNetLib\Tcp\IPAll").TcpPort.Split(",")[0]
            'InstanceVersion' =(Get-ItemProperty
"HKLM:\SOFTWARE\Microsoft\Microsoft SQL Server\$path\Setup").Version
            'InstanceEdition' =(Get-ItemProperty
"HKLM:\SOFTWARE\Microsoft\Microsoft SQL Server\$path\Setup").Edition
            'InstancePatch' =(Get-ItemProperty
"HKLM:\SOFTWARE\Microsoft\Microsoft SQL Server\$path\Setup").PatchLevel
      }
    })
}
Write-Host $($obj | ConvertTo-Json )

Pedro Rocha ***@***.***> escreveu no dia segunda, 29/05/2023 à(s)
10:00:

> you can do a select @@servername
>
> Show you example from a server
> [image: image.png]
> [image: image.png]
>
>
> Guillaume Bougard ***@***.***> escreveu no dia segunda,
> 29/05/2023 à(s) 09:55:
>
>> The name is extracted from the output of SELECT @@version SQL request.
>> So how I can find "MSSQLSERVER" from a SQL request ?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/glpi-project/glpi-agent/discussions/407#discussioncomment-6027755>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AGJFAZ3LUJGLHJBFHWT3FN3XIRQBTANCNFSM6AAAAAAYRCLFXI>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***
>> com>
>>
>
>
> --
> Pedro Rocha
> Faro
> email: ***@***.*** ***@***.***>
> Telf.: +351 935 478 010 <+351%20935%20478%20010>
> linkedin: Pedro Rocha <http://www.linkedin.com/in/pedro-rocha79>
>


-- 
Pedro Rocha
Faro
email: ***@***.*** ***@***.***>
Telf.: +351 935 478 010 <+351%20935%20478%20010>
linkedin: Pedro Rocha <http://www.linkedin.com/in/pedro-rocha79>

**Reply by g-bougard** [reply]
Thank you, but the purpose of the database modules is to permit remote inventory.

**Comment by jpnr79** [comment]
Pedro Rocha ***@***.***> escreveu no dia segunda, 29/05/2023 à(s)
09:43:

> Hello Guillaume
>
> This is the out put you ask, the problem is the "name":"SQL SERVER 2019"
> the real name for the instance is MSSQLSERVER that is the default
>
> [image: image.png]
>
> C:\Program Files\GLPI-Agent>glpi-inventory.bat --partial database
> [info] New partial inventory from
> NSRBDF01.justica.local-2023-05-29-09-26-30 for local0
> {
>    "action": "inventory",
>    "content": {
>       "bios": {
>          "assettag": "8557-7311-0707-8618-3869-1851-25",
>          "bdate": "2019-12-17",
>          "biosserial": "8557-7311-0707-8618-3869-1851-25",
>          "bmanufacturer": "Microsoft Corporation",
>          "bversion": "Hyper-V UEFI Release v4.0",
>          "enclosureserial": "8557-7311-0707-8618-3869-1851-25",
>          "mmodel": "Virtual Machine",
>          "msn": "8557-7311-0707-8618-3869-1851-25",
>          "smanufacturer": "Microsoft Corporation",
>          "smodel": "Virtual Machine",
>          "ssn": "8557-7311-0707-8618-3869-1851-25"
>       },
>       "databases_services": [
>          {
>             "databases": [
>                {
>                   "creation_date": "2003-04-08 09:13:36",
>                   "is_active": true,
>                   "name": "master",
>                   "size": 519,
>                   "update_date": "2023-05-11 15:42:15"
>                },
>                {
>                   "creation_date": "2023-05-10 08:50:46",
>                   "is_active": true,
>                   "name": "tempdb",
>                   "size": 5120,
>                   "update_date": "2023-05-29 09:25:00"
>                },
>                {
>                   "creation_date": "2003-04-08 09:13:36",
>                   "is_active": true,
>                   "name": "model",
>                   "size": 768,
>                   "update_date": "2020-11-12 10:43:19"
>                },
>                {
>                   "creation_date": "2019-09-24 14:21:42",
>                   "is_active": true,
>                   "name": "msdb",
>                   "size": 1061,
>                   "update_date": "2023-03-31 14:07:12"
>                },
>                {
>                   "creation_date": "2020-11-12 12:02:50",
>                   "is_active": true,
>                   "name": "M3Database",
>                   "size": 3648,
>                   "update_date": "2021-05-24 23:56:25"
>                },
>                {
>                   "creation_date": "2022-12-20 09:37:27",
>                   "is_active": true,
>                   "name": "AdventureWorks2019",
>                   "size": 592,
>                   "update_date": "2023-05-11 15:22:07"
>                },
>                {
>                   "creation_date": "2022-03-17 11:08:29",
>                   "is_active": true,
>                   "name": "DBA",
>                   "size": 1024,
>                   "update_date": "2023-02-03 11:23:39"
>                },
>                {
>                   "creation_date": "2023-03-23 15:44:09",
>                   "is_active": true,
>                   "name": "teste_xpto_erro_viste",
>                   "size": 1024,
>                   "update_date": "2023-05-11 15:59:39"
>                }
>             ],
>             "is_active": true,
>             "last_boot_date": "2023-05-10 08:50:42",
>             "manufacturer": "Microsoft",
>             "name": "SQL Server 2019",
>             "port": 1433,
>             "size": 13757,
>             "type": "mssql",
>             "version": "15.0.4236.7"
>          }
>       ],
>       "hardware": {
>          "chassis_type": "Desktop",
>          "memory": 8190,
>          "name": "NSRBDF01",
>          "uuid": "B3CA54D3-3D17-4697-B9CF-A96546846FDA",
>          "vmsystem": "Hyper-V",
>          "winlang": "1033",
>          "winowner": "Windows User",
>          "winprodid": "00429-70000-00000-AA596",
>          "winprodkey": "N69G4-B89J2-4G8F4-WWYCC-J464C",
>          "workgroup": "justica.local"
>       },
>       "versionclient": "GLPI-Inventory_v1.4"
>    },
>    "deviceid": "NSRBDF01.justica.local-2023-05-29-09-26-30",
>    "itemtype": "Computer",
>    "partial": true
> }
>
>
> Pedro Rocha ***@***.***> escreveu no dia segunda, 29/05/2023 à(s)
> 09:25:
>
>> Hello Guillaume
>>
>> I'm going to send you the information, and help to make this part great.
>>
>> I have another question and problem in hands, that is netdiscovery dont
>> resolve name, there are any way to put nsloockup to give the name when
>> netdiscovery dont resolve name by netbios, by arp or snmp?
>>
>> Guillaume Bougard ***@***.***> escreveu no dia segunda,
>> 29/05/2023 à(s) 08:51:
>>
>>> Hi @jpnr79 <https://github.com/jpnr79>
>>>
>>> of course, if you find a bug and have the skill to help us to
>>> understand, we will take it.
>>>
>>> Can you share the output of the following command run from an
>>> administrative console and from the GLPI-Agent agent folder ? Only the
>>> databases_services node is relevant.
>>>
>>> glpi-inventory --partial database
>>>
>>> Then can you share what output you expected or point which values are
>>> wrong ? And finally can you give me a method to catch the right value ?
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/glpi-project/glpi-agent/discussions/407#discussioncomment-6027295>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AGJFAZ5ZYEHOAXKQ4L5HLKDXIRIP5ANCNFSM6AAAAAAYRCLFXI>
>>> .
>>> You are receiving this because you were mentioned.Message ID:
>>> ***@***.***
>>> com>
>>>
>>
>>
>> --
>> Pedro Rocha
>> Faro
>> email: ***@***.*** ***@***.***>
>> Telf.: +351 935 478 010 <+351%20935%20478%20010>
>> linkedin: Pedro Rocha <http://www.linkedin.com/in/pedro-rocha79>
>>
>
>
> --
> Pedro Rocha
> Faro
> email: ***@***.*** ***@***.***>
> Telf.: +351 935 478 010 <+351%20935%20478%20010>
> linkedin: Pedro Rocha <http://www.linkedin.com/in/pedro-rocha79>
>


-- 
Pedro Rocha
Faro
email: ***@***.*** ***@***.***>
Telf.: +351 935 478 010 <+351%20935%20478%20010>
linkedin: Pedro Rocha <http://www.linkedin.com/in/pedro-rocha79>
